### PR TITLE
Feature: support more configuration for l7 health check

### PR DESCRIPTION
**Pull request by dengqian**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/upstream/healthcheck/httpdial.go [hunk 1]
```diff
@@ -43,22 +43,32 @@ func init() {
...
+type CodeRange struct {
+	Start int
+	End   int
+}
+
 type HttpCheckConfig struct {
 	Port    int                `json:"port,omitempty"`
 	Timeout api.DurationConfig `json:"timeout,omitempty"`
 	Path    string             `json:"path,omitempty"`
+	Method  string             `json:"methos,omitempty"`
```
Comment (doujiang24) [inline pkg/upstream/healthcheck/httpdial.go:55]:
```suggestion
	Method  string             `json:"method,omitempty"`
```

**Review by doujiang24 — COMMENTED** [review]
just a minor typo, otherwise LGTM.

File: pkg/upstream/healthcheck/httpdial.go [hunk 2]
```diff
@@ -101,41 +115,70 @@ func (f *HTTPDialSessionFactory) NewSession(cfg map[string]interface{}, host typ
...
+	}
+
+	httpDial.client = &http.Client{
+		Timeout: httpDial.timeout,
 	}
 
-	httpDail.client = &http.Client{
-		Timeout: httpDail.timeout,
+	httpDial.request, err = http.NewRequest(httpCheckConfig.Method, uri.String(), nil)
+	if err != nil {
```
Comment (taoyuanyuan) [inline pkg/upstream/healthcheck/httpdial.go:133]: log err.

File: pkg/upstream/healthcheck/httpdial.go [hunk 3]
```diff
@@ -101,41 +115,70 @@ func (f *HTTPDialSessionFactory) NewSession(cfg map[string]interface{}, host typ
...
+
+	httpDial.Codes = httpCheckConfig.Codes
+
+	log.DefaultLogger.Infof("[upstream] [health check] [httpdial session]  create a health check success for %s", uri.String())
+	return httpDial
+}
+
+func (s *HTTPDialSession) verifyCode(code int) bool {
+	log.DefaultLogger.Infof("get code:%d", code)
+	// default: [200, 400)
```
Comment (taoyuanyuan) [inline pkg/upstream/healthcheck/httpdial.go:149]: How to configure the default config?  [200, 399] ?
maybe [200, 200] is better.

**Review by doujiang24 — APPROVED** [review]

**Review by antJack — APPROVED** [review]
LGTM

**Review by taoyuanyuan — APPROVED** [review]

**Pull request merged**